internal/transport/http2_client.go

```diff
@@ -1025,6 +1027,14 @@ func (t *http2Client) Close(err error) {
 	}
 	t.cancel()
 	t.conn.Close()
+	// Wait for the reader goroutine to exit to ensure all resources are cleaned
+	// up before Close can return.
+	<-t.readerDone
+	if t.keepaliveEnabled {
+		// Wait for the keepAlive goroutine to exit to ensure all resources are cleaned
+		// up before Close can return.
+		<-t.keepAliveDone
+	}
```

Comment:
A single comment might be sufficient.
```suggestion
	// Wait for the reader and keepalive goroutines to exit before returning from here.
	<-t.readerDone
	if t.keepaliveEnabled {
		<-t.keepaliveDone
	}
```

Reply:
Done